File: dotnet/src/Core/Models/Messages/ChatMessage.cs
namespace Agience.Core.Models.Messages
{
    public class ChatMessage
    {
        public string? AuthorRole { get; private set; }
        public string? Content { get; private set; }

        public ChatMessage(string authorRole, string content)
        {
            AuthorRole = authorRole;
            Content = content;
        }

    }
}

File: python/src/core/models/messages/chat_message.py
from typing import Optional
from pydantic import BaseModel, Field


class ChatMessage(BaseModel):
    _author_role: str = Field(alias="author_role")
    _content: str = Field(alias="content")

    def __init__(self, author_role: str, content: str):
        super().__init__(author_role=author_role, content=content)

    @property
    def author_role(self) -> Optional[str]:
        return self._author_role

    @property
    def content(self) -> Optional[str]:
        return self._content

    class Config:
        populate_by_name = True
        frozen = True
